notebooks/243-depth-estimation-videpth/243-depth-estimation-videpth.ipynb

```diff
@@ -0,0 +1,1088 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "98466560-036d-41d2-87f7-957af6f64626",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# Monocular Visual-Inertial Depth Estimation using OpenVINO™\n",
+    "\n",
+    "<p align=\"center\" width=\"100%\">\n",
+    "    <img src=\"https://raw.githubusercontent.com/isl-org/VI-Depth/main/figures/methodology_diagram.png\">    \n",
+    "</p>\n",
+    "\n",
+    "A visual-inertial depth estimation pipeline that integrates monocular depth estimation and visual-inertial odometry to produce dense depth estimates with metric scale has been presented by the authors. The approach consists of three stages: \n",
+    "\n",
+    "1. input processing, where RGB and IMU data feed into monocular depth estimation alongside visual-inertial odometry,\n",
+    "2. global scale and shift alignment, where monocular depth estimates are fitted to sparse depth from VIO in a least-squares manner and \n",
+    "3. learning-based dense scale alignment, where globally-aligned depth is locally realigned using a dense scale map regressed by the ScaleMapLearner (SML). \n",
+    "\n",
+    "The images at the bottom in the diagram above illustrate a VOID sample being processed through our pipeline; from left to right: the input RGB, ground truth depth, sparse depth from VIO, globally-aligned depth, scale map scaffolding, dense scale map regressed by SML, final depth output.\n",
+    "\n",
+    "<p align=\"center\" width=\"100%\">\n",
+    "    <img src=\"https://raw.githubusercontent.com/isl-org/VI-Depth/main/figures/teaser_figure.png\">    \n",
+    "</p>\n",
+    "\n",
+    "We will be consulting this [repo](https://github.com/isl-org/VI-Depth) for the pre-processing, model transformations and basic utility code. A part of it has already been kept as it is in the [utils](utils) directory. At the same time we will learn how to perform [model conversion](https://docs.openvino.ai/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_PyTorch.html) for converting a model in a different format to the standard OpenVINO™ IR model representation *via* another format."
```

Comment:
I noticed that you've chosen to use the .onnx format for your model. 
I'd appreciate it if you could shed some light on your choice of the .onnx format. Additionally, could we also consider adding support for the IR format? I believe having this  option could make it more versatile

Reply:
@pronoym99 I was wondering if you have had a chance to consider it and if you are planning to work on it.

Reply:
@AnishaUdayakumar I did look into the above recommendation of using the IR format to make it more versatile. However, at this point in time I have decided to stick to the concept of pure `.onnx` file download on the fly and its direct compilation and transformation. Later if required, I may raise a PR for supporting the IR format. In my concluding notes, however, I have already mentioned where they are supposed to be found (`.bin` and `.xml` files) if required.